File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Organisms/OverflowMenu/OrgOverflowMenuItem.swift
import Resources

public struct OrgOverflowMenuItem {
    public let text: String
    public let icon: ImageAsset
    public let onTap: () -> Void
    public let isDestructive: Bool

    public init(text: String, icon: ImageAsset, onTap: @escaping () -> Void, isDestructive: Bool = false) {
        self.text = text
        self.icon = icon
        self.onTap = onTap
        self.isDestructive = isDestructive
    }
}

File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/organisms/overflowmenu/OrgOverflowMenuItem.kt
package com.perrystreet.woof.designsystem.atomic.organisms.overflowmenu

import androidx.annotation.DrawableRes
import androidx.compose.runtime.Immutable

@Immutable
data class OrgOverflowMenuItem(
    val text: String,
    @DrawableRes val iconRes: Int,
    val onTap: () -> Unit,
    val isDestructive: Boolean = false,
)
